src/test/scala/com/fulcrumgenomics/alignment/NeedlemanWunschAlignerTest.scala

```diff
@@ -235,4 +243,39 @@ class NeedlemanWunschAlignerTest extends UnitSpec {
     result.cigar.toString() shouldBe "1I11=1I"
     result.score shouldBe 11 - 100 - 1 - 100 - 1
   }
+
+  // Glocal alignment tests
+  "NeedlemanWunschAligner.align(Glocal)" should "align two identical sequences with all matches" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Glocal).align(s("ACGTAACC"), s("ACGTAACC"))
+    assertValidGlocalAlignment(result)
+    result.cigar.toString() shouldBe "8="
+    result.score shouldBe 8
+  }
+
+  it should "correctly align an identical subsequence" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Glocal).align(s("CCGG"), s("AACCGGTT"))
+    assertValidGlocalAlignment(result)
+    result.queryStart shouldBe 1
+    result.targetStart shouldBe 3
+    result.cigar.toString() shouldBe "4="
+    result.score shouldBe 4
+  }
+
+  it should "correctly align an a sub-sequence with a mismatch" in {
+    val result = NeedlemanWunschAligner(1, -1, -3, -1, mode=Glocal).align(s("-------CGCGTCGTATACGTCGTT"), s("AAGATATCGCGTCGTATACGTCGTA"))
```

Comment:
what's with the leading dashes?  Do they get ignored?  Or is this in effect "matches nothing"?  If the latter, can you put "N"s instead.

Reply:
That `s()` function strips out any `-`s and upper-cases the whole string.  The idea being that I can insert `-`s to make it easier to see where the gaps are and how the strings align, but then they're stripped out before being fed into the aligner.

Reply:
I missed the `s` function, sorry.